integration/oasis/stationgraph/station_graph.go

```diff
@@ -1,11 +1,143 @@
 package stationgraph
 
+import (
+	"github.com/Telenav/osrm-backend/integration/oasis/chargingstrategy"
+	"github.com/Telenav/osrm-backend/integration/oasis/stationfinder"
+	"github.com/golang/glog"
+)
+
 type stationGraph struct {
-	g             *graph
-	stationID2Int map[string]uint32
+	g                 *graph
+	stationName2Nodes map[string][]*node
+	stationID2Name    map[uint32]string
+	nodesCount        uint32
+	strategy          chargingstrategy.ChargingStrategyCreator
+}
+
+// NewStationGraph creates station graph from channel
+func NewStationGraph(c chan stationfinder.WeightBetweenNeighbors, currEnergyLevel, maxEnergyLevel float64, strategy chargingstrategy.ChargingStrategyCreator) *stationGraph {
+	sg := &stationGraph{
+		g: &graph{
+			startNodeID: invalidNodeID,
+			endNodeID:   invalidNodeID,
+		},
+		stationName2Nodes: make(map[string][]*node),
+		stationID2Name:    make(map[uint32]string),
+		nodesCount:        0,
+		strategy:          strategy,
+	}
+
+	for item := range c {
+		if item.Err != nil {
+			glog.Errorf("Met error during constructing stationgraph, error = %v", item.Err)
+			return nil
+		}
+
+		for _, neighborInfo := range item.NeighborsInfo {
+			sg.buildNeighborInfoBetweenNodes(neighborInfo, currEnergyLevel, maxEnergyLevel)
+		}
+	}
+
+	return sg.constructGraph()
+}
+
+func (sg *stationGraph) buildNeighborInfoBetweenNodes(neighborInfo stationfinder.NeighborInfo, currEnergyLevel, maxEnergyLevel float64) {
+	for _, fn := range sg.getChargeStationsNodes(neighborInfo.FromName, currEnergyLevel, maxEnergyLevel) {
```

Comment:
What's the meaning of `fn`/`tn` here? `fromNode`/`toNode`? If so, I'll suggest to use full name instead...